etc/kayobe/environments/ci-aio/automated-setup.sh

```diff
@@ -1,109 +1,333 @@
+}
+
+# Run all Kayobe commands in sequence
+run_kayobe_commands() {
+    echo "Running Kayobe commands..."
+    kayobe_control_host_bootstrap
+    kayobe_run_init_playbooks
+    kayobe_overcloud_host_configure
+    kayobe_overcloud_service_deploy
+    kayobe_test_overcloud_vm
+}
+
+# Run Tempest tests
+run_tempest() {
+    echo "Running Tempest tests..."
+    if type apt > /dev/null 2>&1; then
+        sudo apt install -y docker-buildx-plugin
+    fi
+
+    pushd "$BASE_PATH/src/kayobe-config"
     git submodule init
     git submodule update
-    sudo DOCKER_BUILDKIT=1 docker build --build-arg BASE_IMAGE=rockylinux:9 --file .automation/docker/kayobe/Dockerfile --tag kayobe:latest --network host .
-    export KAYOBE_AUTOMATION_SSH_PRIVATE_KEY=$(cat ~/.ssh/id_rsa)
+
+    if ! sudo docker image inspect kayobe:latest > /dev/null 2>&1; then
+        echo "Building Kayobe Automation image"
+        sudo DOCKER_BUILDKIT=1 docker build \
+        --build-arg BASE_IMAGE=rockylinux:9 \
+        --build-arg USE_PYTHON_312=true \
+        --file .automation/docker/kayobe/Dockerfile \
+        --tag kayobe:latest \
+        --network host .
+    else
+        echo "Kayobe Automation image already exists. Skipping build."
+    fi
+
+    KAYOBE_AUTOMATION_SSH_PRIVATE_KEY=$(cat ~/.ssh/id_rsa)
+    export KAYOBE_AUTOMATION_SSH_PRIVATE_KEY
     mkdir -p tempest-artifacts
-    sudo -E docker run --name kayobe-automation --detach -it --rm --network host \
-    -v $(pwd):/stack/kayobe-automation-env/src/kayobe-config -v $(pwd)/tempest-artifacts:/stack/tempest-artifacts \
-    -e KAYOBE_ENVIRONMENT -e KAYOBE_VAULT_PASSWORD -e KAYOBE_AUTOMATION_SSH_PRIVATE_KEY kayobe:latest \
-    /stack/kayobe-automation-env/src/kayobe-config/.automation/pipeline/tempest.sh -e ansible_user=stack
+    
+    activate_kayobe_env
+    sudo -E docker run \
+        --name kayobe-automation \
+        --detach -it --rm \
+        --network host \
+        -v "$(pwd)":/stack/kayobe-automation-env/src/kayobe-config \
+        -v "$(pwd)"/tempest-artifacts:/stack/tempest-artifacts \
+        -e KAYOBE_ENVIRONMENT \
+        -e KAYOBE_VAULT_PASSWORD \
+        -e KAYOBE_AUTOMATION_SSH_PRIVATE_KEY \
+        kayobe:latest \
+        /stack/kayobe-automation-env/src/kayobe-config/.automation/pipeline/tempest.sh \
+        -e ansible_user=stack
+    
+    deactivate
+    echo "Sleeping for 5 minutes to allow Tempest to begin"
     sleep 300
     sudo docker logs -f tempest
-else
-    export KAYOBE_CONFIG_SOURCE_PATH=$BASE_PATH/src/kayobe-config
-    export KAYOBE_VENV_PATH=$BASE_PATH/venvs/kayobe
-    pushd $BASE_PATH/src/kayobe
-    ./dev/overcloud-test-vm.sh
-fi
+    set +x
+    popd
+}
+
+# Generate a Kayobe environment script for quick dev access
+generate_kayobe_env_script() {
+
+cat > $BASE_PATH/kayobe-env.sh <<EOF
+
+export KAYOBE_VAULT_PASSWORD=\$(cat $VAULT_PASSWORD_FILE)
+source $BASE_PATH/src/kayobe-config/kayobe-env --environment ci-aio
+source $BASE_PATH/venvs/kayobe/bin/activate
+
```

Comment:
Might want autocompletion here too, like in beokay?
```suggestion
source <(kayobe complete)

```

Reply:
See previous discussion